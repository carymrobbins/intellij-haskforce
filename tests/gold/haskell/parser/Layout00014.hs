module Layout00014 where

instance Indexed (Pull sh a) where
    Pull ixf _ ! i = ixf i
