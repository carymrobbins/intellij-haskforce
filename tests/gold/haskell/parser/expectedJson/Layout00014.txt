[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":5,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":20}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":8},"srcInfoPoints":[]},"Layout00014"]},null,null]},[],[],[{"InstDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":3,"srcSpanEndLine":5,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":35,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":30},{"srcSpanFilename":"A.hs","srcSpanEndColumn":5,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":5},{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":1}]},null,{"IHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10},"srcInfoPoints":[]},"Indexed"]}]},[{"TyParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":18},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":18},{"srcSpanFilename":"A.hs","srcSpanEndColumn":29,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":28}]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":19},"srcInfoPoints":[]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":19},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":19},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":19},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":23,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":19},"srcInfoPoints":[]},"Pull"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":24},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":24},"srcInfoPoints":[]},"sh"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":27},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":27},"srcInfoPoints":[]},"a"]}]}]}]}]]},[{"InsDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"FunBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":5},"srcInfoPoints":[]},[{"InfixMatch":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"PApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":5},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":5},"srcInfoPoints":[]},"Pull"]}]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},"ixf"]}]},{"PWildCard":{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":14},"srcInfoPoints":[]}}]]},{"Symbol":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":16},"srcInfoPoints":[]},"!"]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":18},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":18},"srcInfoPoints":[]},"i"]}]}],{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":20},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":20}]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":22},"srcInfoPoints":[]},"ixf"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":26},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":26},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":27,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":26},"srcInfoPoints":[]},"i"]}]}]}]}]},null]}]]}]}]]}]]},[]]