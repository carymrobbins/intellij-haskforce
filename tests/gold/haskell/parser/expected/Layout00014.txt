Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('Layout00014')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    INSTANCEDECL
      PsiElement(HaskellTokenType.instance)('instance')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('Indexed')
          ATYPE
            PsiElement(HaskellTokenType.()('(')
            CTYPE
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('Pull')
                ATYPE
                  TYVAR
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('sh')
                ATYPE
                  TYVAR
                    VARID
                      PsiElement(HaskellTokenType.varidRegexp)('a')
            PsiElement(HaskellTokenType.))(')')
      PsiElement(HaskellTokenType.where)('where')
      IDECL
        FUNORPATDECL
          PAT
            QCON
              QCONID
                CONID
                  PsiElement(HaskellTokenType.conidRegexp)('Pull')
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('ixf')
            PsiElement(HaskellTokenType._)('_')
            PsiElement(HaskellTokenType.!)('!')
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('i')
          RHS
            PsiElement(HaskellTokenType.=)('=')
            EXP
              QVAR
                QVARID
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('ixf')
              QVAR
                QVARID
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('i')